docs/model_developers/modules.rst

```diff
@@ -122,3 +122,96 @@ The format for requires in requirements.txt is the folllowing:
  * Or a repository location such as  git+https://github.com/project/python-foo The correct syntax
    to use is then: eggname@git+https://../repository#branch with branch being optional.
 
+
+Working on modules
+==================
+Since modules often have dependencies on other modules, it is common to develop against multiple
+modules (or a project and one or more modules) simultaneously. One might for example need to
+extend a dependent module to add support for some new feature. Because this use case is so common,
+this section will describe how to work on multiple modules simultaneously so that any changes are
+visible to the compiler. This procedure is of course applicable for working on a single module as well.
+
+Setting up the dev environment
+------------------------------
+To set up the development environment for a project, activate your development Python environment and
+install the project with ``inmanta project install``. To set up the environment for a single module,
+run ``inmanta module install`` instead.
+
+The following subsections explain any additional steps you need to take if you want to make changes
+to one of the dependency modules as well.
+
+v1 modules
+^^^^^^^^^^
+Any modules you find in the project's ``modulepath`` after starting from a clean project and setting
+up the development environment are v1 modules. You can make changes to these modules and they will
+be reflected in the next compile. No additional steps are required.
+
+v2 modules
+^^^^^^^^^^
+All other modules are v2 and have been installed by ``inmanta project install`` into the active Python
```

Comment:
Would we allow this? I thought v2 modules would be strictly Python packages, either from a package repo or from an editable install.

Reply:
yes, but an editable install may also result from a download in git form? 

(also: how can the compiler know if it is a V1 or V2 module prior to downloading it?)

Reply:
1. I think this is another thing we never discussed explicitly. According to my view an editable install would always be performed explicitly by the model developer. I thought we wanted to fully move away from the git distribution method for v2, and only support it for backwards compatibility with v1 modules.
2. I don't think it can. I think the repo needs to be cloned to determine this.